engines/dreamweb/dreamweb.cpp

```diff
@@ -396,7 +396,13 @@ void DreamWebEngine::processEvents(bool processSoundEvents) {
 
 Common::Error DreamWebEngine::run() {
 	if (_ttsMan != nullptr) {
-		_ttsMan->setLanguage(Common::getLanguageCode(getLanguage()));
+		Common::String languageString = Common::getLanguageCode(getLanguage());
+		_ttsMan->setLanguage(languageString);
+		if (languageString != "en") {
+			_textEncoding = Common::kDos850;
+		} else {
+			_textEncoding = Common::kUtf8;
+		}
```

Comment:
`kDos850` should work for English as well.
It will definitely not work for Russian though. If I had to take a guess it would be that it might be using `kDos866`. But since this is a fan translation, it may be using something more recent (e.g. `kWindows1251`). You probably want to ask sev. Since he added support for it (in https://github.com/scummvm/scummvm/commit/93dfd39) he might know what encoding is used.

For the checks it would also be better to use the language code rather than string comparisons:
```c+
switch (getLanguage()) {
case Common::RU_RUS:
    _textEncoding = Common::kDos866;
    break;
default:
    _textEncoding = Common::kDos850;
    break;
}
```

Reply:
After checking, the encoding for Russian is `kDos866`
As for Czech, I see this message popping up when I tried to open the game
```
The game in 'dreamweb-cz' seems to be an unknown game variant.

Please report the following data to the ScummVM team at https://bugs.scummvm.org/ along with the name of the game you tried to add and its version, language, etc.:

Matched game IDs for the dreamweb engine: dreamweb-cd

  {"dreamweb.r00", 0, "3b5c87717fc40cc5a5ae19c155662ee3", 152918},
  {"dreamweb.r02", 0, "28458718167a040d7e988cf7d2298eae", 210466},
```

Reply:
> As for Czech, I see this message popping up when I tried to open the game
> What should I do? Do I go post at https://bugs.scummvm.org/?
> 
> ```
> The game in 'dreamweb-cz' seems to be an unknown game variant.
> 
> Please report the following data to the ScummVM team at https://bugs.scummvm.org/ along with the name of the game you tried to add and its version, language, etc.:
> 
> Matched game IDs for the dreamweb engine: dreamweb-cd
> 
>   {"dreamweb.r00", 0, "3b5c87717fc40cc5a5ae19c155662ee3", 152918},
>   {"dreamweb.r02", 0, "28458718167a040d7e988cf7d2298eae", 210466},
> ```
This matches the detection we already have for the Czech version, but you are probably missing the `dreamweb.exe` file that is also used for the detection.

The Czech version download is not the full game. This is a patch that you apply on top of another version. As indicated in https://bugs.scummvm.org/ticket/7078, you can for example use the CD DOS Version (English US) that you can find on https://www.scummvm.org/games/#games-dreamweb and replace some of the files with those in the Czech downloads.

Reply:
Here's what I get printed when setting the encoding to Windows1250
Looks a lot like what is printed on the screen & Google translate recognizes it
```
Neviden...
Obydli samotnych myslenek formuje nasi budoucnost...  Dreamweb.
..  Dreamweb.
Strazena svymi sluhy.......
Miliardy let stara......
Po staleti byla Dreamweb neotresena..........
Nyni ale chce zlo prevzit vladu.........
"Strazci. Pavucina snu se pomalu borti."         
"Sedm sil zla na Zemi se pomalu spojuji v jednu"   
"Pokud se stanou prilis silnymi, Dreamweb bude znicen"    
"Kdo bude ten vykonavatel ?"
"Kdy se to stane?"
"TICHO! Ti, kdoz byli vyvoleni, prichazeji."  
"Pokud odhali svoji silu, pak se stanou prilis silnymi"
"Bylo seme zasazeno?"
"Ano. Mohutne vzrostlo a ma se cile k svetu." 
"Prisel cas a ja ho probudim."
"Tohle je budoucnost.."
"Vim, co ode mne chcete. Myslim, ze vam nyni rozumim."
"Tech sedm se stalo silnejsimi."
"Strasili mne v mych snech."  
"Vim. Maji moc nad sedmi misty pavuciny snu - Dreamweb."
"Museji byt zniceni?"    
"Ano."
"Citim blizkost vsech sedmi. Ktery z nich ma byt prvni?"     
"Jmenuje se  Crane."
"Je nejslabsi ze vsech sedmi."
"Jak ho mam najit?"
"Je velmi blizko. Opatri si zbran a zacni ho hledat." 
"Cas se nam kvapem krati, bratre...."
```